### PR TITLE
Remove filter query(fq) input param from read(), update() and delete() functions

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject

Related issue: https://github.com/ballerina-platform/ballerina-standard-library/issues/3469

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [x] Added tests

### CODE REVIEW

**Review by daneshk — APPROVED** [review]

File: ballerina/tests/company_client.bal [hunk 1]
```diff
@@ -52,8 +52,8 @@ client class CompanyClient {
         return <Company>check self.persistClient.runReadByKeyQuery(Company, key, include);
     }
 
-    remote function read(map<anydata>? filter = (), CompanyRelations[] include = []) returns stream<Company, Error?> {
-        stream<anydata, sql:Error?>|Error result = self.persistClient.runReadQuery(Company, filter, include);
+    remote function read(CompanyRelations[] include = []) returns stream<Company, Error?> {
```
Comment (kaneeldias) [inline ballerina/tests/company_client.bal:55]: Remove `filter` param from read methods

File: ballerina/tests/company_client.bal [hunk 2]
```diff
@@ -52,8 +52,8 @@ client class CompanyClient {
...
+        stream<anydata, sql:Error?>|Error result = self.persistClient.runReadQuery(Company, include);
```
Comment (kaneeldias) [inline ballerina/tests/company_client.bal:56]: Remove `filter` param from `runReadQuery()`

File: ballerina/tests/company_client.bal [hunk 3]
```diff
@@ -70,12 +70,12 @@ client class CompanyClient {
         }
     }
 
-    remote function update(record {} 'object, map<anydata> filter) returns Error? {
-        _ = check self.persistClient.runUpdateQuery('object, filter);
+    remote function update(Company 'object) returns Error? {
```
Comment (kaneeldias) [inline ballerina/tests/company_client.bal:73]: Remove `filter` param and change type of 1st param to entityName from `update()` method

File: ballerina/tests/company_client.bal [hunk 4]
```diff
@@ -70,12 +70,12 @@ client class CompanyClient {
...
+        _ = check self.persistClient.runUpdateQuery('object);
     }
 
-    remote function delete(map<anydata> filter) returns Error? {
-        _ = check self.persistClient.runDeleteQuery(filter);
+    remote function delete(Company 'object) returns Error? {
```
Comment (kaneeldias) [inline ballerina/tests/company_client.bal:77]: Remove filter param and change type of 1st param to entityName from update() method

File: ballerina/tests/company_client.bal [hunk 5]
```diff
@@ -70,12 +70,12 @@ client class CompanyClient {
         }
     }
 
-    remote function update(record {} 'object, map<anydata> filter) returns Error? {
-        _ = check self.persistClient.runUpdateQuery('object, filter);
+    remote function update(Company 'object) returns Error? {
+        _ = check self.persistClient.runUpdateQuery('object);
```
Comment (kaneeldias) [inline ballerina/tests/company_client.bal:74]: Remove filter `param` from `runUpdateQuery()`

File: ballerina/tests/company_client.bal [hunk 6]
```diff
@@ -70,12 +70,12 @@ client class CompanyClient {
...
     }
 
-    remote function delete(map<anydata> filter) returns Error? {
-        _ = check self.persistClient.runDeleteQuery(filter);
+    remote function delete(Company 'object) returns Error? {
+        _ = check self.persistClient.runDeleteQuery('object);
```
Comment (kaneeldias) [inline ballerina/tests/company_client.bal:78]: Remove `filter` param from `runDeleteQuery()` and replace with `'object`

File: ballerina/tests/employee_client.bal [hunk 7]
```diff
@@ -78,29 +78,18 @@ client class EmployeeClient {
...
-            error? e =  from Employee employee in employeeStream
-                do {
-                    if employee.company is Company {
-                        check companyClient->update(companyEntity, {"id": (<Company>employee.company).id});
-                    }
-                };
-            if e is error {
-                return <Error>e;
-            }
+            check companyClient->update(companyEntity);
```
Comment (kaneeldias) [inline ballerina/tests/employee_client.bal:87]: Replace instances of open records with Entity

File: ballerina/tests/employee_client.bal [hunk 8]
```diff
@@ -78,29 +78,18 @@ client class EmployeeClient {
...
-            // TODO: replace this with more optimized code after adding support for advanced queries
-            error? e =  from Employee employee in employeeStream
-                do {
-                    if employee.company is Company {
-                        check companyClient->update(companyEntity, {"id": (<Company>employee.company).id});
-                    }
-                };
-            if e is error {
-                return <Error>e;
-            }
```
Comment (kaneeldias) [inline ballerina/tests/employee_client.bal:98]: Delete these code segments for assocations. Not necessary

**Pull request merged**